### PR TITLE
Add utilities to pass transactions through contexts

### PR DESCRIPTION
**Pull request by c-du**
## Description

Add the utilities for passing transactions through contexts. Will show an example in the blob store
From Gorman's https://github.com/stackrox/stackrox/pull/5225

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: pkg/postgres/conn.go [hunk 1]
```diff
@@ -41,6 +41,10 @@ func (c *Conn) Exec(ctx context.Context, sql string, args ...interface{}) (pgcon
 	ctx, cancel := contextutil.ContextWithTimeoutIfNotExists(ctx, defaultTimeout)
 	defer cancel()
 
+	if tx, ok := TxFromContext(ctx); ok {
+		return tx.Exec(ctx, sql, args...)
+	}
+
 	ct, err := c.Conn.Exec(ctx, sql, args...)
 	if err != nil {
 		incQueryErrors(sql, err)
```
Comment (dashrews78) [inline pkg/postgres/conn.go:50]: Should we update the query error counts for the transaction from context too?

**Review by dashrews78 — APPROVED** [review]
Just the one question on error counts which will be simple if we choose to do it.  Otherwise looks good.

**Review by connorgorman — CHANGES_REQUESTED** [review]
We should add tests for this

File: tools/generate-helpers/pg-table-bindings/testgraphtables/testchild1/store_tx_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
+//go:build sql_integration
+
+package postgres
```
Comment (c-du) [inline tools/generate-helpers/pg-table-bindings/testgraphtables/testchild1/store_tx_test.go:3]: This is piggy back the existing test suite I can add this to all the stores eventually.

**Review by connorgorman — APPROVED** [review]
Looks good!

File: pkg/postgres/conn.go [hunk 3]
```diff
@@ -81,8 +109,31 @@ func (c *Conn) QueryRow(ctx context.Context, sql string, args ...interface{}) *R
...
+	ctx, cancel := contextutil.ContextWithTimeoutIfNotExists(ctx, defaultTimeout)
+	defer cancel()
+
+	if tx, ok := TxFromContext(ctx); ok {
+		rows, err = tx.CopyFrom(ctx, tableName, columnNames, rowSrc)
+	} else {
+		rows, err = c.PgxPoolConn.CopyFrom(ctx, tableName, columnNames, rowSrc)
+	}
+	if err != nil {
+		incQueryErrors("copyfrom", err)
```
Comment (connorgorman) [inline pkg/postgres/conn.go:136]: do you think we should just make incQueryErrors not do anything if err == nil. That should make this a one liner?

Comment (c-du) [reply pkg/postgres/conn.go:136]: Yes, good point. I will make this small change.

**Pull request merged**